packages/frontend/src/components/wallet-migration/Sender/MigrationWithQrCode.jsx

```diff
@@ -0,0 +1,104 @@
+import { QRCodeSVG } from 'qrcode.react';
+import React, { useState, useEffect } from 'react';
+import { Translate } from 'react-localize-redux';
+import styled from 'styled-components';
+
+import { encryptAccountsData, generateSalt } from '../../../utils/encoding';
+import FormButton from '../../common/FormButton';
+import Modal from '../../common/modal/Modal';
+import { getAccountsData } from '../WalletMigration';
+
+
+const Container = styled.div`
+    padding: 15px 0;
+    text-align: center;
+    margin: 0 auto;
+
+    @media (max-width: 360px) {
+        padding: 0;
+    }
+
+    @media (min-width: 500px) {
+        padding: 56px 24px 24px;
+    }
+
+    .title {
+        font-weight: 800;
+        font-size: 20px;
+        margin-top: 40px;
+    }
+
+    .qrcode-background {
+        background-color: white;
+        padding: 5px;
+        margin-top: 20px;
+        margin-bottom: 20px;
+    }
+`;
+
+const ButtonsContainer = styled.div`
+    text-align: center;
+    width: 100% !important;
+    display: flex;
+    padding: 12px 24px 24px;
+    border-top: 1px solid #EDEDED;
+`;
+
+const StyledButton = styled(FormButton)`
+    width: calc((100% - 16px) / 2);
+
+    &:last-child {
+        margin-left: 16px !important;
+    }
+`;
+
+const MigrationWithQrCode = ({ accounts, pinCode, onClose }) => {
+    const [qrCode, setQrCode] = useState('');
+
+    useEffect(() => {
+        const generateQrCode = async () => {
+            const salt = generateSalt(12);
+            const accountsData = await getAccountsData(accounts);
+            const encryptData = encryptAccountsData(accountsData, pinCode, salt);
```

Comment:
`pinCode` is a six digit number and too easy to brute force for encrypting the payload in a machine-decipherable QR code. We need something with higher entropy here, ideally a base 64 string, BIP 39 phrase, or equivalent.